hbase-client/src/main/java/org/apache/hadoop/hbase/client/Admin.java

```diff
@@ -1582,7 +1591,7 @@ default Future<Void> modifyTableAsync(TableName tableName, TableDescriptor td)
    * @return the result of the async modify. You can use Future.get(long, TimeUnit) to wait on the
    *         operation to complete
    */
-  Future<Void> modifyTableAsync(TableDescriptor td) throws IOException;
+  Future<Void> modifyTableAsync(TableDescriptor td, boolean reopenRegions) throws IOException;
```

Comment:
I think we need to provide a new overload, otherwise this change is not compatible for existing users of this method.

Reply:
@bbeaudreault To clarify - you mean, the external clients who might be directly calling the modifyTableAsync(TableDescriptor td) method ? As i have modified the other callers in the codebase here.

Reply:
Correct. Admin, AsyncAdmin, Table, AsyncTable are public APIs, so we need to retain backwards compatibility on them. Breaking interface changes (such as changing a signature or deleting a method) require 2 major version releases, with a migration path between. So if we wanted to change this signature, we need to provide an overload and then mark the original as `@Deprecated`. The deprecated method can't be removed until at least hbase 4.0.

I don't think we need to deprecate the original method right now since reopenRegions is sort of an advanced usage, but just providing the context -- we do need to create an overload.

Reply:
Got it. Thanks @bbeaudreault for the explanation. I will make the change.